src/Parallel/Pin.cpp

```diff
@@ -82,6 +130,14 @@ cpu_set_t seissol::parallel::Pinning::getFreeCPUsMask() const {
   cpu_set_t freeMask{};
   CPU_ZERO(&freeMask);
 
+  if (not parsedFreeCPUsMask.empty()) {
+    const auto localProcessor = MPI::mpi.sharedMemMpiRank();
+    for (auto& cpu : parsedFreeCPUsMask[localProcessor]) {
```

Comment:
Breaks here if parsedFreeCPUsMask.size() <= localProcessor

Reply:
You could also allow round-robin assignment. E.g. if the free list is "39-40" and you have two processes then you could pin both threads to CPUs 39 and 40 and let the OS scheduler decide the rest.

Reply:
> You could also allow round-robin assignment.

I can see the point but I would like to leave the implementation as it is if nobody is against. Here is my reasoning. 

I think that people (mostly developers) are going to use this option when the automatic free cpus pinning failed for some reason. In this case, a person is desperate and wants to describe pinning  as clear as possible. In such a scenario, he/she is going to be ok to spend a bit of time on typing but be 100% sure about the locations of free cpus.

Reply:
Sure, maybe better that way.